File: app/src/main/java/com/stathis/elmepaunivapp/ui/dashboard/DashboardActivity.java
package com.stathis.elmepaunivapp.ui.dashboard;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;

import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.stathis.elmepaunivapp.abstraction.AbstractActivity;
import com.stathis.elmepaunivapp.listeners.activity_listeners.DashboardActivityClickListener;
import com.stathis.elmepaunivapp.ui.dashboard.model.DashboardOption;
import com.stathis.elmepaunivapp.ui.department.DepartmentActivity;
import com.stathis.elmepaunivapp.ui.professors.ProfessorsActivity;
import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.ui.announcements.AnnouncementActivity;
import com.stathis.elmepaunivapp.ui.chatbot.ChatBotActivity;
import com.stathis.elmepaunivapp.ui.students.StudentsActivity;
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity;

public class DashboardActivity extends AbstractActivity implements View.OnClickListener, DashboardActivityClickListener {

    private RecyclerView dashboardOptions;
    private ImageView about;
    private FloatingActionButton chatbot_btn;
    private DashboardViewModel viewModel;

    public DashboardActivity() {
        super(R.layout.activity_dashboard);
    }

    @Override
    public void initial() {
        viewModel = new ViewModelProvider(this).get(DashboardViewModel.class);

        chatbot_btn = findViewById(R.id.fab_chatbot);
        about = findViewById(R.id.about);
        dashboardOptions = findViewById(R.id.dashboard_options_recview);
    }

    @Override
    public void running() {
        dashboardOptions.setAdapter(viewModel.dashboardAdapter);
        viewModel.initListener(this);
        viewModel.displayData();

        about.setOnClickListener(this);
        chatbot_btn.setOnClickListener(this);
    }

    @Override
    public void stopped() {}

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.fab_chatbot:
                startActivity(new Intent(DashboardActivity.this, ChatBotActivity.class));
                break;
            case R.id.about:
                viewModel.showDialog(this);
                break;
        }
    }

    @Override
    public void goToAnnouncementScreen(DashboardOption dashboardOption) {
        startActivity(new Intent(DashboardActivity.this, AnnouncementActivity.class));
    }

    @Override
    public void goToDepartmentScreen(DashboardOption dashboardOption) {
        startActivity(new Intent(DashboardActivity.this, DepartmentActivity.class));
    }

    @Override
    public void goToProfessorScreen(DashboardOption dashboardOption) {
        startActivity(new Intent(DashboardActivity.this, ProfessorsActivity.class));
    }

    @Override
    public void goToStudentsScreen(DashboardOption dashboardOption) {
        startActivity(new Intent(DashboardActivity.this, StudentsActivity.class));
    }

    @Override
    public void learnMore() {
        startActivity(new Intent(DashboardActivity.this, WebviewActivity.class).putExtra("URL", "https://mst.hmu.gr/ypiresies/mobile-epharmogh-tmhmatos/"));
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/dashboard/DashboardActivity.kt
package com.stathis.elmepaunivapp.ui.dashboard

import android.content.Intent
import androidx.lifecycle.ViewModelProvider
import com.stathis.elmepaunivapp.R
import com.stathis.elmepaunivapp.abstraction.ElmepaActivity
import com.stathis.elmepaunivapp.callbacks.DashboardClickListener
import com.stathis.elmepaunivapp.ui.announcements.AnnouncementsActivity
import com.stathis.elmepaunivapp.ui.chatbot.ChatbotActivity
import com.stathis.elmepaunivapp.ui.dashboard.model.DashboardOption
import com.stathis.elmepaunivapp.ui.department.DepartmentActivity
import com.stathis.elmepaunivapp.ui.professors.ProfessorActivity
import com.stathis.elmepaunivapp.ui.students.StudentsActivity
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity
import kotlinx.android.synthetic.main.activity_dashboard.*

class DashboardActivity : ElmepaActivity(R.layout.activity_dashboard) {

    private lateinit var viewModel : DashboardActivityViewModel

    override fun init() {
        viewModel = ViewModelProvider(this).get(DashboardActivityViewModel::class.java)
    }

    override fun startOps() {
        dashboard_options_recview.adapter = viewModel.adapter

        viewModel.bindList(object : DashboardClickListener {
            override fun dashboardItemClicked(option: DashboardOption) {
                when(option.drawable){
                    R.drawable.ic_announcement -> startActivity(Intent(this@DashboardActivity, AnnouncementsActivity::class.java))
                    R.drawable.ic_books -> startActivity(Intent(this@DashboardActivity, DepartmentActivity::class.java))
                    R.drawable.ic_student -> startActivity(Intent(this@DashboardActivity, StudentsActivity::class.java))
                    R.drawable.ic_teacher -> startActivity(Intent(this@DashboardActivity, ProfessorActivity::class.java))
                    else -> Unit
                }
            }

            override fun learnMore() {
                startActivity(Intent(this@DashboardActivity, WebviewActivity::class.java)
                    .putExtra("URL","https://mst.hmu.gr/ypiresies/mobile-epharmogh-tmhmatos/"))
            }
        })

        fab_chatbot.setOnClickListener{
            startActivity(Intent(this@DashboardActivity, ChatbotActivity::class.java))
        }

        about.setOnClickListener{
            viewModel.showDialog(this)
        }
    }

    override fun stopOps(){}
}